spec/request/api/v1/auth/confirmation_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+describe 'Auth', type: :request do
+  let(:user) { build(:user) }
+  let(:sign_up_params) do
+    {
+      name: user.name,
+      email: user.email,
+      password: user.password,
+      password_confirmation: user.password
+    }
+  end
+
+  describe 'POST api/v1/auth/confirmation' do
+    before do
+      post api_v1_user_registration_path sign_up_params
+      @confirmation_url = email_confirmation_url last_email
+    end
+
+    context 'when the request is succesful' do
+      before do
+        get @confirmation_url
+      end
+
+      it 'is expected a successful response' do
+        expect(response).to have_http_status(:found)
+      end
+
+      it 'is expected that user is confirmed' do
+        expect(User.last.confirmed?).to eq true
```

Comment:
In here you are testing that the User is confirmed, but you are not certain that was the previous action that confirmed it. You could do that like this:
```
expect { get @confirmation_url }.to change { User.last.confirmed? }.from(false).to(true)
```